ts/Stock/Indicators/DMI/DMIIndicator.ts

```diff
@@ -0,0 +1,398 @@
+/* *
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type {
+    DMIOptions,
+    DMIParamsOptions
+} from './DMIOptions';
+import type DMIPoint from './DMIPoint';
+import type IndicatorValuesObject from '../IndicatorValuesObject';
+import type LineSeries from '../../../Series/Line/LineSeries';
+import MultipleLinesMixin from '../../../Mixins/MultipleLines.js';
+import SeriesRegistry from '../../../Core/Series/SeriesRegistry.js';
+const {
+    seriesTypes: {
+        sma: SMAIndicator
+    }
+} = SeriesRegistry;
+import U from '../../../Core/Utilities.js';
+const {
+    correctFloat,
+    extend,
+    isArray,
+    merge
+} = U;
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * The Directional Movement Index (DMI) series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.dmi
+ *
+ * @augments Highcharts.Series
+ */
+class DMIIndicator extends SMAIndicator {
+    /**
+     * Directional Movement Index (DMI).
+     * This series requires the `linkedTo` option to be set and should
+     * be loaded after the `stock/indicators/indicators.js` file.
+     *
+     * @sample stock/indicators/dmi
+     *         DMI indicator
+     *
+     * @extends      plotOptions.sma
+     * @since        next
+     * @product      highstock
+     * @excluding    allAreas, colorAxis, joinBy, keys, navigatorOptions,
+     *               pointInterval, pointIntervalUnit, pointPlacement,
+     *               pointRange, pointStart, showInNavigator, stacking
+     * @requires     stock/indicators/indicators
+     * @requires     stock/indicators/dmi
+     * @optionparent plotOptions.dmi
+     */
+    public static defaultOptions: DMIOptions = merge(SMAIndicator.defaultOptions, {
+        /**
+         * @excluding index
+         */
+        params: {
+            index: void 0, // DMI has unchangeable index
```

Comment:
The algorithm is strictly based on specific `high`, `low` and `close` values. I excluded it from API, but the `index` field is still displayed in a popup in Stock Tools. Should we correct it in Stock Tools (bug/feature)? Or can it be somehow handled here?

Reply:
Mentioned issue is reported here: #15362

Reply:
Without assigning `undefined` here, it would inherit the value from SMA and that value would be displayed in Stock Tools popup. I decided to assign `undefined` to clearly let everyone know that the index is no needed. Furthermore, we will fix it in Stock Tools popup, probably just by filtering out `undefined` values or similarly.

Reply:
I would leave it as it is and change/improve in the future if needed.

Reply:
I would leave it as it is and change/improve in the future if needed (unless you have any objections).